### PR TITLE
fe-insuree E2E workflow 1

### PR DESCRIPTION
**Pull request by Blue-B-code**
# Description

This branch adds comprehensive Cypress end-to-end tests designed to validate the full user experience. The tests cover all key application modules, ensuring that user workflows function correctly and consistently across the platform. This helps maintain high quality and reliability throughout the application.

# Type of Change

- [ ] Feature
- [ ] Bug fix
- [x] Chore (Refactor, Docs, CI/CD)
- [ ] Other, please specify

## Related Issue(s) / Task(s)
- [ ] These PRs needs to be merged first before this one  : 
- [ ] Relates to [link to github PR], this needs to be merged before [link to github PR]
- [ ] External reference (e.g., Jira):

## Demo

[Capture vidéo du 2026-02-20 10-24-20.webm](https://github.com/user-attachments/assets/71025e31-f433-48e2-a6a2-99524238e41f)

## Checklist
- [ ] Unit tests added/modified
- [ ] I18n / translation handled

### CODE REVIEW

File: cypress/e2e/insuree.cy.js [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  }
+  cy.get('[data-cy="InsureeMainMenu"]').click();
+  cy.get('[href="/front/insuree/family"]').click();
+}
+
+// =====================================================
+// 🔹 FORM HELPERS
+// =====================================================
+export function fillInsureeForm(insuree) {
+  cy.get('[data-cy="insuree-chf-id-input"] input').clear().type(insuree.chfId);
```
Comment (weilu) [inline cypress/e2e/insuree.cy.js]: Please reuse existing helper cy commands: `cy.enterMuiInput` Example use case: https://github.com/openimis/openimis-dist_dkr/blob/develop/cypress/e2e/cash-transfer.cy.js#L108

Definition: https://github.com/openimis/openimis-dist_dkr/blob/develop/cypress/support/commands.js#L602-L609

Note that the helper is defined based on input labels rather than UI element attributes. As a general principle, E2E tests should be written from the user perspective as much as possible – so use labels and references as what the user sees, not invisible element ids and data-cy attributes which are not visible to the user. This approach also wouldn't require frontend changes as cited in the dependency PRs.

File: cypress/e2e/insuree.cy.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+  cy.get('[data-cy="insuree-email-input"] input').clear().type(insuree.email);
+  cy.get('[data-cy="insuree-passport-input"] input').clear().type(insuree.passport);
+
+  // Pickers
+  cy.get('[data-cy="insuree-profession-picker"]').click();
+  cy.get(`[data-value="${insuree.profession}"]`).click();
+  cy.get('[data-cy="insuree-education-picker"]').click();
+  cy.get(`[data-value="${insuree.education}"]`).click();
+  cy.get('[data-cy="insuree-type-of-id-picker"]').click();
+  cy.get(`[data-value='\"${insuree.typeOfId}\"']`).click();
```
Comment (weilu) [inline cypress/e2e/insuree.cy.js]: Use `chooseMuiSelect`, defined here: https://github.com/openimis/openimis-dist_dkr/blob/develop/cypress/support/commands.js#L611-L615

File: cypress/support/commands.js [hunk 3]
```diff
@@ -685,3 +685,30 @@ Cypress.Commands.add('getItemCount', (itemName) => {
...
+  clickCreateInsuree,
+  clickSave,
+  goToFamiliesPage,
+  goToFamilyForm,
+  goToInsureesPage,
+  verifyFamilyExists,
+  verifyInsureeExists,
+  selectSearcherRow,
+  addExistingInsureeIntoFamily
+} from '../e2e/insuree.cy';
```
Comment (weilu) [inline cypress/support/commands.js]: commands.js contains utility functions and commands used by E2E tests, whereas insuree.cy contains actual E2E tests, so this import relationship is reversed. Please define commands in this file. Please make sure to first check if there are reusable commands already defined before defining new ones.

**Review by weilu — CHANGES_REQUESTED** [review]
Please refactor element selectors to use labels/visible text rather than IDs or data-cy attributes, and reuse our existing Cypress helpers where applicable (see inline comments for specifics).

E2E tests should validate what users actually see and interact with. Label-based selection also catches accessibility regressions – if a label is missing or misassociated, the test fails, as it should. This is a good principle to keep in mind: "[The more your tests resemble the way your software is used, the more confidence they can give you.](https://testing-library.com/docs/guiding-principles)"

Reserve attribute-based selectors only for edge cases like duplicate labels or non-visible elements.

File: cypress/e2e/insuree.cy.js (hunk 1 above)
Comment (Blue-B-code) [reply cypress/e2e/insuree.cy.js]: thanks for your review

File: cypress/e2e/insuree.cy.js (hunk 2 above)
Comment (Blue-B-code) [reply cypress/e2e/insuree.cy.js]: fixed

File: cypress/support/commands.js (hunk 3 above)
Comment (Blue-B-code) [reply cypress/support/commands.js]: fixed

File: cypress/support/commands.js [hunk 4]
```diff
@@ -685,3 +685,30 @@ Cypress.Commands.add('getItemCount', (itemName) => {
...
+Cypress.Commands.add('fillInsureeForm', fillInsureeForm);
+Cypress.Commands.add('clickCreateInsuree', clickCreateInsuree);
+Cypress.Commands.add('clickSave', clickSave);
+Cypress.Commands.add('goToFamiliesPage', goToFamiliesPage);
+Cypress.Commands.add('goToFamilyForm', goToFamilyForm);
+Cypress.Commands.add('goToInsureesPage', goToInsureesPage);
+Cypress.Commands.add('verifyFamilyExists', verifyFamilyExists);
+Cypress.Commands.add('verifyInsureeExists', verifyInsureeExists);
+Cypress.Commands.add('selectSearcherRow', selectSearcherRow);
+Cypress.Commands.add('addExistingInsureeIntoFamily', addExistingInsureeIntoFamily);
```
[the remark this answers is not in the document]
Comment (Blue-B-code) [reply cypress/support/commands.js]: fixed

File: cypress/e2e/insuree.cy.js [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+export function goToFamiliesPage() {
+  cy.get('[data-cy="InsureeMainMenu"]').click();
+  cy.get('[href="/front/insuree/families"]').click();
+}
+
+export function goToFamilyForm(familyChfId) {
+  if (familyChfId) {
+    goToFamiliesPage();
+    verifyFamilyExists({ chfId: familyChfId });
+    selectSearcherRow(familyChfId);
```
[the remark this answers is not in the document]
Comment (Blue-B-code) [reply cypress/e2e/insuree.cy.js]: fixed

File: cypress.config.js [hunk 6]
```diff
@@ -51,7 +51,7 @@ module.exports = defineConfig({
   e2e: {
     projectId: "q6gc25", // Cypress Cloud, needed for recording
     baseUrl: 'http://localhost',
-    defaultCommandTimeout: 15000,
+    defaultCommandTimeout: 200000,
```
[the remark this answers is not in the document]
Comment (Blue-B-code) [reply cypress.config.js]: thanks for your review; fixed

File: cypress/support/commands.js [hunk 7]
```diff
@@ -611,13 +611,62 @@ Cypress.Commands.add('enterMuiInput', (label, value, inputTag='input') => {
 Cypress.Commands.add('chooseMuiSelect', (label, value) => {
   cy.contains('label', label)
     .siblings('.MuiInputBase-root')
-    .find('[role="button"]')
-    .click()
+    .click(); 
+
+  cy.get('body')
+    .contains('li[role="option"]', value, { timeout: 10000 })
```
Comment (weilu) [inline cypress/support/commands.js:617]: This selector's scope is potentially too big and can accidentally match other `li`s. Can it be narrowed down?

File: cypress/support/commands.js [hunk 8]
```diff
@@ -611,13 +611,62 @@ Cypress.Commands.add('enterMuiInput', (label, value, inputTag='input') => {
...
+      .click();
+      cy.get('.MuiYearCalendar-button')
+      .contains(year)
+      .click();
+      cy.get('[aria-label="year view is open, switch to calendar view"]')
+      .should('be.visible')
+      .click();
+    }
+    if (month) {
+      cy.get('[aria-label="year view is open, switch to calendar view"]')
```
Comment (weilu) [inline cypress/support/commands.js:651]: `'[aria-label="year view is open, switch to calendar view"]'` is used 3 times. Consider extracting into constant/variable.

File: cypress/e2e/insuree.cy.js [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+    cy.contains('tr', head.chfId)
+      .within(() => {
+        cy.contains('button', 'Delete').click();
+      });
+    cy.contains('button', 'Delete family and members').click();
+  }
+};
+
+// Tests
+describe.only('Family & Insuree Workflow', () => {
```
Comment (weilu) [inline cypress/e2e/insuree.cy.js:174]: Remove `.only, otherwise only this test will run

File: cypress/e2e/insuree.cy.js [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+
+    cy.save();
+
+    // Add / Remove member
+    Family.addMember(insurees.head, insurees.member);
+    Family.removeMember(insurees.head, insurees.member);
+
+    // Cleanup
+    Insuree.delete(insurees.member);
+    Family.delete(insurees.head);
```
Comment (weilu) [inline cypress/e2e/insuree.cy.js:214]: Consider moving this to `after` or `afterEach` to ensure cleanup even when tests were to fail.

**Review by weilu — CHANGES_REQUESTED** [review]
Looking much better! Great progress. See additional in-line comments

**Review by vaneck237 — APPROVED** [review]

File: cypress/e2e/insuree.cy.js (hunk 10 above)
Comment (Blue-B-code) [reply cypress/e2e/insuree.cy.js:214]: moving this in after or afterEach could also fail if the insuree was not created successfully ?

File: cypress/support/commands.js (hunk 7 above)
Comment (Blue-B-code) [reply cypress/support/commands.js:617]: This poses no problem.

File: cypress/e2e/insuree.cy.js [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+  },
+
+  fillForm: (insuree) => {
+    cy.enterMuiInput('Insurance No.', insuree.chfId, "input");
+    cy.enterMuiInput('Last Name', insuree.lastName);
+    cy.enterMuiInput('Given Names', insuree.givenNames);
+    cy.enterMuiInput('Phone', insuree.phone);
+    cy.enterMuiInput('Email', insuree.email);
+
+    cy.chooseMuiDatePicker('Birth Date', insuree.dobDay);
```
[the remark this answers is not in the document]
Comment (Blue-B-code) [reply cypress/e2e/insuree.cy.js:73]: fixed in https://github.com/openimis/openimis-dist_dkr/pull/92

File: cypress/e2e/insuree.cy.js [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+  }
+};
+
+export const familyData = {
+  location: 'R1D1M1V1 Rachla',
+  familyType: 'Household',
+  confirmationNo: 'CONF-001',
+  confirmationType: 'Municipality',
+  address: 'Douala, Cameroun'
+};
```
[the remark this answers is not in the document]
Comment (Blue-B-code) [reply cypress/e2e/insuree.cy.js:40]: fixed in https://github.com/openimis/openimis-dist_dkr/pull/92

File: cypress/e2e/insuree.cy.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (Blue-B-code) [reply cypress/e2e/insuree.cy.js:174]: fixed in https://github.com/openimis/openimis-dist_dkr/pull/92

File: cypress/support/commands.js [hunk 13]
```diff
@@ -611,13 +611,62 @@ Cypress.Commands.add('enterMuiInput', (label, value, inputTag='input') => {
...
+      .click();
+      cy.get('.MuiYearCalendar-button')
+      .contains(month)
+      .click();
+    }
+    cy.get('[role="gridcell"]')
+    .contains(day)
+    .should('be.visible')
+    .click();
 })
```
[the remark this answers is not in the document]
Comment (Blue-B-code) [reply cypress/support/commands.js:662]: fixed in https://github.com/openimis/openimis-dist_dkr/pull/92

File: cypress/e2e/insuree.cy.js [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+    cy.contains('button', 'Search').click({force: true});
+    cy.contains(head.lastName);
+    cy.contains(head.givenNames);
+    cy.contains(head.chfId);
+  },
+
+  addMember: (head, member) => {
+    Family.goToForm(head);
+    cy.contains('button', 'Add existing').click()
+    cy.contains('label', 'Insurance No.').type(member.chfId)
```
[the remark this answers is not in the document]
Comment (Blue-B-code) [reply cypress/e2e/insuree.cy.js:145]: fixed in https://github.com/openimis/openimis-dist_dkr/pull/92

File: cypress/e2e/insuree.cy.js (hunk 10 above)
Comment (Blue-B-code) [reply cypress/e2e/insuree.cy.js:214]: fixed in https://github.com/openimis/openimis-dist_dkr/pull/92

File: cypress/e2e/insuree.cy.js (hunk 9 above)
Comment (Blue-B-code) [reply cypress/e2e/insuree.cy.js:174]: fixed in https://github.com/openimis/openimis-dist_dkr/pull/92

File: cypress/support/commands.js (hunk 8 above)
Comment (Blue-B-code) [reply cypress/support/commands.js:651]: fixed in https://github.com/openimis/openimis-dist_dkr/pull/92

**Comment by weilu** [comment]
@Blue-B-code please use the branch `develop` as the base branch for future PRs. I've updated it for this one. Kindly resolve merge conflicts. 

Also, going forward please keep commits for addressing comments of a given PR within the same PR. #92 contains changes of its own and those for addressing comments in this PR – it makes it very difficult for reviewers.

**Comment by Blue-B-code** [comment]
the goal of this PR is now maintained and developed in https://github.com/openimis/openimis-dist_dkr/pull/92
